### PR TITLE
feat: edit business infos

### PR DESCRIPTION
**Pull request by elysee15**
Fixes SC-2556
This PR adds an option to update the business informations

https://www.awesomescreenshot.com/video/6706970?key=cb210e86ce7dab66973ffc1bbe7886c0

Updated screen
- For a matters of UX I moved the submit and cancel button
Screen Shot 2022-01-07 at 22 01 07

### CODE REVIEW

**Comment by bbengfort** [comment]
Looks good so far! I played around with it a bit and everything seemed to work as I expected. All the validation was exactly right!

A couple of minor notes on the modal:

Screen Shot 2022-01-07 at 15 14 55

- Could we please change the title to "Edit Business Information". 
- Could we add a description after the title "Please enter extended and basic details about the business entity. Note that the IVMS 101 data contains the legal details and company name information." 

Is it possible to add more rows to the multi-select? Perhaps we could try making it 8 rows if so?

**Comment by bbengfort** [comment]
@elysee15 I started to work through the merge conflicts to rebase from main, but after looking at a few of them I figured it would be better if you did it so I didn't mess up anything you were currently working on. Would you mind rebasing this PR?

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        </Field.Select>
+                    </Col>
+                    <Form.Text>Please select the entity category that most closely matches your organization.
+                    </Form.Text>
+                </Form.Group>
+                <Form.Group as={Row} className="mb-2" controlId='vasp_categories'>
+                    <Form.Label column sm="12" className="fw-normal">
+                        VASP Category
+                    </Form.Label>
+                    <Col sm="8">
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js:78]: I think this should be `<Col sm="12">` to match the label and other columns; I'm seeing the following:

Screen Shot 2022-01-11 at 15 38 59

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                        <Field.Select register={register} multiple name={`vasp_categories`}>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js:79]: My request before was to add more vertical height to the select; perhaps that got misinterpreted for the column size? Is it possible to add a `data-size="8"` attribute to this field, e.g. [this stackoverflow](https://stackoverflow.com/a/39630641/488917) answer? If not, let's go ahead and leave it as is, resetting the column width to 12.

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/index.js [hunk 3]
```diff
@@ -11,58 +11,7 @@ import Geographic from './components/Geographic';
...
 import { downloadFile } from 'helpers/api/utils';
 import classNames from 'classnames';
-import { actionType, useModal } from 'contexts/modal';
-import { useSelector } from 'react-redux';
-import { getAllReviewNotes } from 'redux/selectors';
-import VaspDocument from '../VaspDocument';
-import { pdf } from '@react-pdf/renderer';
-import NProgress from 'nprogress'
-
-export const BasicDetailsDropDown = ({ isNotPendingReview, vasp }) => {
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/BasicDetails/index.js:21]: Thank you for reorganizing this component!

**Review by bbengfort — COMMENTED** [review]
@elysee15 thank you for resolving those comments; things are looking good - I just have one more change request in the code, then we'll get things merged.

In terms of tests - if we'd like to create a story for tests of this feature; the payload tests I did were:

1. Must have a valid website
2. Must have a valid date
3. Cannot set business category to nil
4. Cannot have empty list of VASP categories
5. Can submit 1 VASP category
6. Can submit multiple VASP categories 

All worked as expected!

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js (hunk 2 above)
Comment (elysee15) [reply web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BusinessInfosForm.js:79]: solved using the props `htmlSize` from react-bootstrap

**Review by bbengfort — APPROVED** [review]
Looks great!

**Pull request merged**